server/schedule/operator_controller.go

```diff
@@ -367,11 +371,15 @@ func (oc *OperatorController) GetOperatorStatus(id uint64) *OperatorWithStatus {
 	return oc.opRecords.Get(id)
 }
 
-func (oc *OperatorController) removeOperatorLocked(op *operator.Operator) {
+func (oc *OperatorController) removeOperatorLocked(op *operator.Operator) bool {
 	regionID := op.RegionID()
-	delete(oc.operators, regionID)
-	oc.updateCounts(oc.operators)
-	operatorCounter.WithLabelValues(op.Desc(), "remove").Inc()
+	if cur := oc.operators[regionID]; cur == op {
```

Comment:
I understand the bug here caused by different operators having the same region ID.  The check here does not guarantees deleting the correct operator, I think.

Reply:
Here is comparing two pointers. Why cannot it guarantee deleting the correct operator? Could you give an example?

Reply:
My mistake, I didn't notice the comparison. I thought it just judges the existence.